[GTK][WPE] CGroupMemoryController::getCgroupFileValue not stack if not numerical value is scanned from the file input (e.g: max)

psaavedra opened the pull request:
#### a5f69d0f997171aeead106dcd8fb85c53ebb3d88
<pre>
[GTK][WPE] CGroupMemoryController::getCgroupFileValue not stack if not numerical value is scanned from the file input (e.g: max)
<a href="https://bugs.webkit.org/show_bug.cgi?id=273931">https://bugs.webkit.org/show_bug.cgi?id=273931</a>

Reviewed by Carlos Alberto Lopez Perez.

Improves the CGroupMemoryController::getCgroupFileValue by ensuring it
handles non-numeric values (e.g., max).

Fixes TestNetworkProcessMemoryPressure API test (webkit.org/b/263016).

* Source/WebKit/UIProcess/linux/MemoryPressureMonitor.cpp:
(WebKit::CGroupMemoryController::getCgroupFileValue):
* Tools/TestWebKitAPI/glib/TestExpectations.json:

Canonical link: <a href="https://commits.webkit.org/278599@main">https://commits.webkit.org/278599@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f8e9b6184c2bebf3c29cfe034885dd9a974a30cc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/51009 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30308 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3335 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/54266 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/1698 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/53310 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36596 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1365 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/41545 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/53107 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27946 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43950 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22664 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/25283 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/1205 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/9449 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/47261 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/1279 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55860 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/26116 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/1175 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/48960 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/27367 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/44021 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/48083 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11171 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/28245 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/27098 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

clopez commented on Source/WebKit/UIProcess/linux/MemoryPressureMonitor.cpp:
```diff
@@ -438,8 +439,20 @@ size_t CGroupMemoryController::getCgroupFileValue(FILE *file)
+    char rawValue[VALUE_BUFFER_SIZE + 1];
+    int res = fscanf(file, "%" STRINGIFY(VALUE_BUFFER_SIZE) "[^\n]", rawValue);
+
+    if (res < 1)
```
the variable res is not really needed, we can check the return code from `fscanf` in the `if()` ifself

psaavedra replied:
valid point.  PR just updated according with this suggestion.

psaavedra committed:
[GTK][WPE] CGroupMemoryController::getCgroupFileValue not stack if not numerical value is scanned from the file input (e.g: max)
https://bugs.webkit.org/show_bug.cgi?id=273931

Reviewed by Carlos Alberto Lopez Perez.

Improves the CGroupMemoryController::getCgroupFileValue by ensuring it
handles non-numeric values (e.g., max).

Fixes TestNetworkProcessMemoryPressure API test (webkit.org/b/263016).

* Source/WebKit/UIProcess/linux/MemoryPressureMonitor.cpp:
(WebKit::CGroupMemoryController::getCgroupFileValue):
* Tools/TestWebKitAPI/glib/TestExpectations.json:

Canonical link: https://commits.webkit.org/278599@main
